extensions/ql-vscode/test/pure-tests/sarif-processing.test.ts

```diff
@@ -595,9 +623,14 @@ describe('SARIF processing', () => {
     expect(msg.startsWith('Error when processing SARIF result')).to.be.true;
   }
 
+  function expectNoParsingError(result: { errors: string[] | undefined }) {
```

Comment:
The result returned by `extractAnalysisAlerts` is always guaranteed to return `errors`. I'm not sure why we're dealing with `undefined` here.

Isn't the check just `expect(result.errors.length).to.equal(0)`?

Reply:
Yes...undefined isn't an option here. I'll change that.

I was using `expect(array.length, array.join()).to.equal(0);` so that the second parameter to expect would print outt he actual array if there was an error. It was helpful earlier on while I was writing the test. And I decided to keep it since it's generally useful.